src/front/glsl/functions.rs

```diff
@@ -1058,6 +1058,71 @@ impl Parser {
         });
     }
 
+    fn arg_type_walker(
```

Comment:
could we drop a comment here explaining what it does? looks to be not simply walking but also populating expressions